From c6bda37bb225921eb7dc89cd8c7faac7ce870fca Mon Sep 17 00:00:00 2001
From: Simone Gasparini <simone.gasparini@gmail.com>
Date: Tue, 2 Feb 2021 23:06:18 +0100
Subject: [PATCH] [cmake] suppress deprecated cuda sm warnings

---
 CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9735bc2..f39f2fe 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,6 +16,7 @@ option(PopSift_USE_GRID_FILTER "Switch off grid filtering to massively reduce co
 option(PopSift_USE_NORMF "The __normf function computes Euclidean distance on large arrays. Fast but stability is uncertain." OFF)
 option(PopSift_NVCC_WARNINGS "Switch on several additional warning for CUDA nvcc" OFF)
 option(PopSift_USE_TEST_CMD "Add testing step for functional verification" OFF)
+option(PopSift_NO_DEPRECATED_CUDA_SM_WARNINGS "Suppress warnings about soon to be deprecated cuda SM" ON)
 option(BUILD_SHARED_LIBS "Build shared libraries" ON)
 
 if(PopSift_USE_POSITION_INDEPENDENT_CODE AND NOT MSVC)
@@ -105,6 +106,10 @@ endif()
 
 set(CUDA_SEPARABLE_COMPILATION ON)
 
+if(PopSift_NO_DEPRECATED_CUDA_SM_WARNINGS)
+  list(APPEND CUDA_NVCC_FLAGS "-Wno-deprecated-gpu-targets")
+endif()
+
 if(UNIX AND NOT APPLE)
   list(APPEND CUDA_NVCC_FLAGS         "-Xcompiler;-rdynamic")
   # set(CUDA_NVCC_FLAGS       "${CUDA_NVCC_FLAGS};-Xptxas;-v")
-- 
2.29.2.windows.2

