diff --git a/cmake/FindDependencies.cmake b/cmake/FindDependencies.cmake
index 05084200..5fc1cf32 100644
--- a/cmake/FindDependencies.cmake
+++ b/cmake/FindDependencies.cmake
@@ -27,11 +27,11 @@ find_package(Boost ${COLMAP_FIND_TYPE} COMPONENTS
 
 find_package(Eigen3 ${COLMAP_FIND_TYPE})
 
-find_package(FreeImage ${COLMAP_FIND_TYPE})
+find_package(freeimage ${COLMAP_FIND_TYPE})
 
-find_package(Metis ${COLMAP_FIND_TYPE})
+find_package(metis ${COLMAP_FIND_TYPE})
 
-find_package(Glog ${COLMAP_FIND_TYPE})
+find_package(glog ${COLMAP_FIND_TYPE})
 if(DEFINED glog_VERSION_MAJOR)
   # Older versions of glog don't export version variables.
   list(APPEND COLMAP_COMPILE_DEFINITIONS GLOG_VERSION_MAJOR=${glog_VERSION_MAJOR})
@@ -43,7 +43,7 @@ find_package(SQLite3 ${COLMAP_FIND_TYPE})
 set(OpenGL_GL_PREFERENCE GLVND)
 find_package(OpenGL ${COLMAP_FIND_TYPE})
 
-find_package(Glew ${COLMAP_FIND_TYPE})
+find_package(GLEW ${COLMAP_FIND_TYPE})
 
 find_package(Git)
 
@@ -65,7 +65,7 @@ if(CGAL_ENABLED)
     set(CGAL_DO_NOT_WARN_ABOUT_CMAKE_BUILD_TYPE TRUE)
     # We do not use CGAL data. This prevents an unnecessary warning by CMake.
     set(CGAL_DATA_DIR "unused")
-    find_package(CGAL ${COLMAP_FIND_TYPE})
+    find_package(cgal ${COLMAP_FIND_TYPE})
 endif()
 
 if(CGAL_FOUND)
diff --git a/src/colmap/math/CMakeLists.txt b/src/colmap/math/CMakeLists.txt
index f27655f9..b34d1955 100644
--- a/src/colmap/math/CMakeLists.txt
+++ b/src/colmap/math/CMakeLists.txt
@@ -44,7 +44,7 @@ COLMAP_ADD_LIBRARY(
         Eigen3::Eigen
     PRIVATE_LINK_LIBS
         Boost::boost
-        metis
+        metis::metis
 )
 
 COLMAP_ADD_TEST(
diff --git a/src/colmap/sfm/CMakeLists.txt b/src/colmap/sfm/CMakeLists.txt
index 3383a73f..08f8b7d3 100644
--- a/src/colmap/sfm/CMakeLists.txt
+++ b/src/colmap/sfm/CMakeLists.txt
@@ -41,6 +41,7 @@ COLMAP_ADD_LIBRARY(
         colmap_scene
         colmap_estimators
         colmap_util
+        Eigen3::Eigen
     PRIVATE_LINK_LIBS
         colmap_geometry
         colmap_image
