diff --git a/src/cmake/Helpers.cmake b/src/cmake/Helpers.cmake
index 9c6f192a9..b4de88b5e 100644
--- a/src/cmake/Helpers.cmake
+++ b/src/cmake/Helpers.cmake
@@ -19,13 +19,14 @@ function(alicevision_add_library library_name)
 
   # Generate Windows versioning information
   if(MSVC)
+    add_compile_definitions(NOGDI)
     set(ALICEVISION_INSTALL_VERSION_MAJOR ${ALICEVISION_VERSION_MAJOR})
     set(ALICEVISION_INSTALL_VERSION_MINOR ${ALICEVISION_VERSION_MINOR})
     set(ALICEVISION_INSTALL_VERSION_REVISION ${ALICEVISION_VERSION_REVISION})
     set(ALICEVISION_INSTALL_NAME ${library_name})
     set(ALICEVISION_INSTALL_LIBRARY 1)
     configure_file(
-      "${CMAKE_SOURCE_DIR}/src/cmake/version.rc.in"
+      "${aliceVision_SOURCE_DIR}/src/cmake/version.rc.in"
       "${CMAKE_CURRENT_BINARY_DIR}/${library_name}_version.rc"
       @ONLY
     )
@@ -154,13 +155,14 @@ function(alicevision_add_software software_name)
 
   # Generate Windows versioning information
   if(MSVC)
+    add_compile_definitions(NOGDI)
     set(ALICEVISION_INSTALL_VERSION_MAJOR ${ALICEVISION_SOFTWARE_VERSION_MAJOR})
     set(ALICEVISION_INSTALL_VERSION_MINOR ${ALICEVISION_SOFTWARE_VERSION_MINOR})
     set(ALICEVISION_INSTALL_VERSION_REVISION 0)
     set(ALICEVISION_INSTALL_NAME ${software_name})
     set(ALICEVISION_INSTALL_LIBRARY 0) # software
     configure_file(
-      "${CMAKE_SOURCE_DIR}/src/cmake/version.rc.in"
+      "${aliceVision_SOURCE_DIR}/src/cmake/version.rc.in"
       "${CMAKE_CURRENT_BINARY_DIR}/${software_name}_version.rc"
       @ONLY
     )
